tests/common/test_gen_traffic_pattern_config.py

```diff
@@ -43,7 +45,14 @@ def test_gen_traffic_pattern_host_group(self, tp_hostfile):
             [['node0', 'node5'], ['node6', 'node4'], ['node7', 'node3'], ['node1', 'node2']],
             [['node0', 'node6'], ['node7', 'node5'], ['node1', 'node4'], ['node2', 'node3']]
         ]
-        self.assertEqual(gen_traffic_pattern_host_group(hostx, pattern), expected_host_group)
+        self.assertEqual(gen_traffic_pattern_host_groups(hostx, pattern), expected_host_group)
+        # Test for pattern_config file
+        temp_dir = tempfile.TemporaryDirectory()
+        test_config_path = '{}/test_pattern_config.txt'.format(temp_dir.name)
```

Comment:
why not create tempfile.NamedTemporaryFile directly?

Reply:
If it will be part of results, please add to documentation for this output

Reply:
updated and add a description in overview.md.